src/interface.rs

```diff
@@ -232,33 +233,38 @@ impl<'a> Interface<'a> {
 
             write!(screen, "{}{}", fg, bg).unwrap();
 
-            let command_line_index = self.command_line_index(index as i16);
-
-            write!(
-                screen,
-                "{}{}",
-                cursor::Goto(
-                    1,
-                    (command_line_index as i16 + result_top_index as i16) as u16
-                ),
-                Interface::truncate_for_display(
-                    command,
-                    &self.input.command,
-                    width,
-                    highlight,
-                    fg,
-                    self.debug
+            let command_display = Interface::truncate_for_display(
+                command,
+                &self.input.command,
+                width,
+                highlight,
+                fg,
+                self.debug,
+            );
+
+            let mut lines_count = 0;
+
+            for line in command_display.lines() {
+                write!(
+                    screen,
+                    "{}{}",
+                    cursor::Goto(
+                        1,
+                        (line_offset as i16 + lines_count as i16 + result_top_index as i16) as u16
+                    ),
+                    line
                 )
-            )
-            .unwrap();
+                .unwrap();
+                lines_count += 1;
+            }
 
             if command.last_run.is_some() {
                 write!(
                     screen,
                     "{}",
                     cursor::Goto(
                         width - 9,
-                        (command_line_index as i16 + result_top_index as i16) as u16
+                        (line_offset as i16 + result_top_index as i16) as u16
```

Comment:
I think some logic got lost here around `is_screen_view_bottom`?

Reply:
Indeed, I pushed another commit to fix this.